core/src/main/scala/shapeless/implicit.scala

```diff
@@ -0,0 +1,76 @@
+package shapeless
+
+/**
+ * Wraps a `T` made of implicit values.
+ *
+ * E.g. looking for an implicit `Implicit[A :: B :: HNil]`, will
+ * look up for implicits `A` and `B`. If both are found, it succeeds and the implicit
+ * `A` and `B` are available via `value`. Same thing for longer HLists.
+ *
+ * Looking for an `Implicit[A :+: B :+: CNil]` will look up for an implicit `A` first,
+ * else for an implicit `B` if it fails. If either one of them is found, it is
+ * available wrapped in a `A :+: B :+: CNil`, via `value`. Same thing with longer coproducts.
+ *
+ * One can also look for types having a `Generic` instance. Their generic representation
+ * will be looked up as above, then converted to the right type with `Generic`.
+ *
```

Comment:
Is there an example of looking for types with a `Generic` instance? I'm wondering if this overlaps with things like `WitnessWith` or the "typelevel-CPS" pattern in `Generic1` and `IsH/CCons1`?